openai-proxy/py_proxy/proxy.py

```diff
@@ -138,6 +140,133 @@ def proxy():
+2. say it was unable to find the answer to your question
+
+Your job is to do the following:
+1. If the question is a process lookup and there is a valid answer, say "The answer can be found in search results [{citations}]", followed by a 1 sentence summary of the answer. Do not use more than 1 sentence.
+2. If the answer  says it's unable to answer the question, or there is no information for the question, say "Unable to answer the question based on the search results. Please look at the top results to find relevant references"
+3. Otherwise return the answer unmodified
+
+If citations are surrounded by a dollar sign and curly braces, e.g. [${1}], change it to only be a number surrounded by square brackets, i.e. [1]
+
+Examples:
+Question: what is the part number I need for a service kit for my Reba RL A5?
+
+System generated answer:
+The answer can be found in search result(s) [1]. The part number for the service kit for a Reba RL A5 is 00.4315.032.650.
+
+Answer:
+The answer can be found in search result(s) [1]. The part number for the service kit for a Reba RL A5 is 00.4315.032.650.
+----
+Question: Are there different color lower leg options for my SID?
+
+
+System generated answer:
+The answer can be found in search result(s) [no citation given].
+The search results do not provide information on different color options for the lower legs of a SID suspension fork. For accurate information, please refer to the product catalog or contact the manufacturer directly.
+
+Answer:
+Unable to answer the question based on the search results. Please look at the top results to find relevant references
+
+----
+
+Question: How do I adjust sag on my super deluxe coil?
+
+System generated answer:
+The answer can be found in search result(s) [4].
+To adjust sag on your Super Deluxe Coil, follow these steps: 
+1. Install the coil spring and spring retainer.
+2. Adjust the spring preload adjuster until the coil spring contacts the spring retainer. Ensure there is no vertical play between the coil spring and the retainer.
+3. Do not exceed 5 mm (or five full turns of rotation) on the spring preload adjuster as this will damage the shock.
+4. If more than 5 turns are necessary to achieve proper sag, use a higher weight spring.
+5. If your target sag percentage is not achieved, spring preload adjustment and/or coil spring replacement must be performed.
+
+Answer:
+The answer can be found in search result [4]. To adjust the sag on your Super Deluxe Coil, you will need to install a coil spring and spring retainer, followed by some additional adjustments.
+
+"""
+
+    question = request.json.get('question')
+    answer = request.json.get('answer')
+    prompt = f"""
+        
+    Current question and generated answer:
+    {question}
+
+    Answer:
+    {answer}
+        """
+    messages = [
+        {"role": "system", "content": summarize_prompt},
+        {"role": "user", "content": prompt}
+    ]
+    open_ai_result = make_openai_call(messages)
```

Comment:
Need try-except around this one, as below?

Reply:
Probably should for all these APIs. I haven't put in too much thought because we need to move this out of here anyway. Also our UI will need to handle the errors better